tokio-macros/src/entry.rs

```diff
@@ -178,14 +180,14 @@ impl Configuration {
 
         let flavor = self.flavor.unwrap_or(self.default_flavor);
         let worker_threads = match (flavor, self.worker_threads) {
-            (F::CurrentThread, Some((_, worker_threads_span))) => {
+            (F::CurrentThread | F::Local, Some((_, worker_threads_span))) => {
                 let msg = format!(
                     "The `worker_threads` option requires the `multi_thread` runtime flavor. Use `#[{}(flavor = \"multi_thread\")]`",
                     self.macro_name(),
                 );
                 return Err(syn::Error::new(worker_threads_span, msg));
             }
-            (F::CurrentThread, None) => None,
+            (F::CurrentThread | F::Local, None) => None,
             (F::Threaded, worker_threads) if self.rt_multi_thread_available => {
```

Comment:
Please add logic to `rt_multi_thread_available` to detect whether `--cfg tokio_unstable` is set and emit a hard error if it is not.

Reply:
Sorry, I just saw your comment.  Is https://github.com/tokio-rs/tokio/commit/4363203e67bb39edf8f959d88fd4940a0771c688 more what you had in mind?

Reply:
This is harder than it appears because [`RUSTFLAGS` is not passed to proc macros](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#target-applies-to-host).  I guess we have to _emit_ a `cfg` check, so it actually can't live in either of these places.